api/src/lib/webhonc/index.ts

```diff
@@ -0,0 +1,161 @@
+import path from "node:path";
+import type { LibSQLDatabase } from "drizzle-orm/libsql";
+import WebSocket from "ws";
+import type { z } from "zod";
+import * as schema from "../../db/schema.js";
+import logger from "../../logger.js";
+import { resolveServiceArg } from "../../probe-routes.js";
+import {
+  executeProxyRequest,
+  handleFailedRequest,
+  handleSuccessfulRequest,
+} from "../proxy-request/index.js";
+import { WsMessageSchema } from "../types.js";
+import { resolveUrlQueryParams } from "../utils.js";
+import { getWebHoncConnectionId, setWebHoncConnectionId } from "./store.js";
+
+export function connectToWebhonc(
+  host: string,
+  db: LibSQLDatabase<typeof schema>,
+  wsConnections: Set<WebSocket>,
+) {
+  const protocol = host.startsWith("localhost") ? "ws" : "wss";
+  const socket = new WebSocket(`${protocol}://${host}/ws`);
+
+  socket.onopen = () => {
+    logger.debug(`Connected to the webhonc service at ${host}`);
+  };
+
+  socket.onmessage = async (event) => {
+    logger.debug("Received message from the webhonc service:", event.data);
+    await handleMessage(event, wsConnections, db);
```

Comment:
```suggestion
    try {
      await handleMessage(event, wsConnections, db);
    } catch (err) {
      logger.debug("Webhonc handleMessage had an error", err);
    }
```

this can throw, might be nice to print the error. could also see if it's an instance of a zod error (not sure what the thrown error looks like?) and format the error string accordingly